app/models/subnet.rb

```diff
@@ -1,17 +1,37 @@
 require 'ipaddr'
+
 class Subnet < ActiveRecord::Base
+  def self.default_sti_class
+    'Subnet::Ipv4' if self.name == 'Subnet'
+  end
+
+  IP_FIELDS = [:network, :mask, :gateway, :dns_primary, :dns_secondary, :from, :to]
```

Comment:
Probably yes, let's open a Redmine for this. If I or somebody else finds the time we can improve this. The code in this PR might not be perfect, but i think it's an improvement. I definitely don't have the time to refactor this even more...